internal/git/repository.go

```diff
@@ -0,0 +1,195 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+)
+
+type Repository struct {
+	// git config
+	syncBranchName string
+	repoURL        string
+	localPath      string
+
+	// auth config
+	auth       *http.BasicAuth
+	authorName string
+
+	// internal state
+	baseBranchName string // main or master
+	repo           *git.Repository
+	workTree       *git.Worktree
+	syncRef        *plumbing.Reference
+}
```

Comment:
nit - I would break these into other structs for clarity but not required. If I find myself demarcating structs by comments, I usually do this.

```suggestion
type Repository struct {
	git         gitConfig
	auth        authConfig
	internal    internalState
}

// gitConfig contains the repository configuration relating to git.
type gitConfig struct {
	syncBranchName string
	repoURL        string
	localPath      string
}

// authConfig contains the repository configuration for auth.
type authConfig struct {
	auth       *http.BasicAuth
	authorName string
}

type internalState struct {
	baseBranchName string // main or master
	repo           *git.Repository
	workTree       *git.Worktree
	syncRef        *plumbing.Reference
}
```

Reply:
discussed orally.
It is nice to do but I prefer for now to keep a simpler structure